code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,175 @@
+//Bureaucracy machine!
+//Simply set this up in the hopline and you can serve people based on ticket numbers
+
+/obj/machinery/ticket_machine
+	name = "Ticket machine"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticketmachine"
+	desc = "Please, take a ticket!."
+	density = FALSE
+	maptext_height = 26
+	maptext_width = 32
+	maptext_x = 7
+	maptext_y = 10
+	layer = HIGH_OBJ_LAYER
+	var/ticket_number = 0 //Increment the ticket number whenever the HOP presses his button
+	var/current_number = 0 //What customer are we currently serving?
+	var/max_number = 999 //To stop the text going fucky. At this point, you need to refill it.
+	var/cooldown = 50 //Small cooldown, stops the clown from immediately breaking it.
+	var/ready = TRUE
+	var/id = "ticket_machine_default" //For buttons
+
+/obj/machinery/ticket_machine/emag_act(mob/user) //Emag the ticket machine to dispense burning tickets, as well as randomize its customer number to destroy the HOP's mind.
+	if(obj_flags & EMAGGED)
+		return
+	to_chat(user, "<span class='warning'>You overload [src]'s bureaucratic logic circuitry to its MAXIMUM setting.</span>")
+	ticket_number = rand(0,999)
+	obj_flags |= EMAGGED
+
+/obj/machinery/ticket_machine/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/machinery/ticket_machine/proc/increment()
+	if(current_number >= ticket_number)
+		return
+	playsound(src, 'sound/misc/announce_dig.ogg', 50, 0)
+	say("Next customer, please!")
+	current_number ++ //Increment the one we're serving.
+	update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+
+/obj/machinery/button/ticket_machine
+	name = "increment ticket counter"
+	desc = "Use this button when you've served a customer to tell the next one to come forward."
+	device_type = /obj/item/assembly/control/ticket_machine
+	req_access = list()
+	id = "ticket_machine_default"
+
+/obj/item/assembly/control/ticket_machine
+	name = "ticket machine controller"
+	desc = "A remote controller for the HOP's ticket machine."
+	var/obj/machinery/ticket_machine/linked //To whom are we linked?
+
+/obj/item/assembly/control/ticket_machine/Initialize()
+	. = ..()
+	find_machine()
+
+/obj/item/assembly/control/ticket_machine/proc/find_machine() //Locate the one to which we're linked
+	for(var/obj/machinery/ticket_machine/ticketsplease in GLOB.machines)
+		if (ticketsplease.id == id)
+			linked = ticketsplease
+	if(linked)
+		return TRUE
+	else
+		return FALSE
+
+/obj/item/assembly/control/ticket_machine/activate()
+	if(cooldown)
+		return
+	if(!linked)
+		if(!find_machine()) //No linked ticket machine, find a new one
```

Comment:
I would avoid redoing this every time, since it does loop the machine list, maybe require hitting it with a multitool and just have it output a failure message and give that a reasonable cooldown before it loops again